### PR TITLE
Focus trap in poll creation dialog

### PR DESCRIPTION
**Pull request by kerryarchibald**
Fixes: https://github.com/vector-im/element-web/issues/20281

https://user-images.githubusercontent.com/3055605/154710570-40109661-6c14-4203-b76d-416fdd466bf5.mov

---
Here's what your changelog entry will look like:

## ✨ Features
 * Focus trap in poll creation dialog ([\#7847](https://github.com/matrix-org/matrix-react-sdk/pull/7847)). Fixes vector-im/element-web#20281. Contributed by @kerryarchibald.

Preview: https://pr7847--matrix-react-sdk.netlify.app
⚠️ Do you trust the author of this PR? Maybe this build will steal your keys or give you malware. Exercise caution. Use test accounts.

### CODE REVIEW

File: src/components/views/elements/PollCreateDialog.tsx [hunk 1]
```diff
@@ -185,18 +191,21 @@ export default class PollCreateDialog extends ScrollableBaseModal<IProps, IState
...
                 onChange={this.onQuestionChange}
                 usePlaceholderAsHint={true}
                 disabled={this.state.busy}
+                autoFocus={this.state.autoFocusTarget === FocusTarget.Topic}
             />
             <h2>{ _t("Create options") }</h2>
             {
                 this.state.options.map((op, i) => <div key={`option_${i}`} className="mx_PollCreateDialog_option">
                     <Field
+                        id={`option_${i}`}
```
Comment (kerryarchibald) [inline src/components/views/elements/PollCreateDialog.tsx:208]: Using the autogenerated field ids means the tests rely on running order of tests, add these to have stable ids

File: src/components/views/elements/PollCreateDialog.tsx [hunk 2]
```diff
@@ -35,10 +35,14 @@ interface IProps extends IDialogProps {
     editingMxEvent?: MatrixEvent;  // Truthy if we are editing an existing poll
 }
 
+enum FocusTarget {
+    Topic, NewOption
```
Comment (t3chguy) [inline src/components/views/elements/PollCreateDialog.tsx:39]:
```suggestion
    Topic,
    NewOption,
```

File: src/components/views/elements/PollCreateDialog.tsx (hunk 1 above)
Comment (t3chguy) [reply src/components/views/elements/PollCreateDialog.tsx:208]: ID should be scoped to this component, so it doesn't clash

**Review by t3chguy — APPROVED** [review]
LGTM otherwise

**Pull request merged**